server/mscalendar/availability_test.go

```diff
@@ -399,11 +403,13 @@ func TestRetrieveUsersToSyncIndividually(t *testing.T) {
 		m := New(env, "").(*mscalendar)
 		jobSummary := &StatusSyncJobSummary{}
 
-		_, _, err := m.retrieveUsersToSync([]*store.UserShort{}, jobSummary, true)
+		_, _, err := m.retrieveUsersToSyncUsingGoroutines(ctx, []*store.UserShort{}, jobSummary, concurrency)
 		require.ErrorIs(t, errNoUsersNeedToBeSynced, err)
 	})
 
 	t.Run("user reminders and status disabled", func(t *testing.T) {
+		ctx := context.TODO()
```

Comment:
Nit: Should there be a test that checks if cancelling `ctx` returns an error? If we have that test, this can become a `context.Background()`.

Reply:
What would be a correct way of testing a context being cancelled? I've tried some stuff but since this runs in goros results with gomock are not consistent, and I'm not sure if using `.AnyTimes()` is a good solution here.

Reply:
You can pass a cancelled context to `retrieveUsersToSyncUsingGoroutines`. This is not the same as waiting for the timeout, but good enough.